spark/src/main/scala/io/delta/tables/DeltaTable.scala

```diff
@@ -562,6 +562,203 @@ class DeltaTable private[tables](
+    cloneAtVersion(version, target, replace, properties = Map.empty)
+  }
+
+  /**
+   * Clone a DeltaTable at a specific version to a given destination to mirror the existing
+   * table's data and metadata.
+   * 
+   * An example would be
+   * {{{
+   *  io.delta.tables.DeltaTable.cloneAtVersion(5, "/some/path")
+   * }}}
+   *
+   * @param version The version of this table to clone from.
+   * @param target The path or table name to create the clone.
+   *
+   * @since 3.3.0
+   */
+  def cloneAtVersion(version: Long, target: String): DeltaTable = {
+    cloneAtVersion(version, target, replace = false)
+  }
+
+   /**
+   * Clone a DeltaTable at a specific timestamp to a given destination to mirror the existing
+   * table's data and metadata.
+   *
+   * Timestamp can be of the format yyyy-MM-dd or yyyy-MM-dd HH:mm:ss.
+   *
+   * Specifying properties here means that the target will override any properties with the same key
+   * in the source table with the user-defined properties.
+   *
+   * An example would be
+   * {{{
+   *  io.delta.tables.DeltaTable.cloneAtTimestamp(
+   *   "2019-01-01",
+   *   "/some/path",
+   *   true,
+   *   Map("foo" -> "bar"))
+   * }}}
+   *
+   * @param timestamp The timestamp of this table to clone from.
+   * @param target The path or table name to create the clone.
+   * @param replace Whether to replace the destination with the clone command.
+   * @param properties The table properties to override in the clone.
+   *
+   * @since 3.3.0
+   */
+  def cloneAtTimestamp(
+      timestamp: String,
+      target: String,
+      replace: Boolean,
+      properties: Map[String, String]): DeltaTable = {
+    executeClone(table, target, replace, properties, timestampAsOf = Some(timestamp))
+  }
+
+  /**
+   * Clone a DeltaTable at a specific timestamp to a given destination to mirror the existing
+   * table's data and metadata.
+   * 
+   * Timestamp can be of the format yyyy-MM-dd or yyyy-MM-dd HH:mm:ss.
+   * 
+   * An example would be
+   * {{{
+   *  io.delta.tables.DeltaTable.cloneAtTimestamp("2019-01-01", "/some/path", true)
+   * }}}
+   *
+   * @param timestamp The timestamp of this table to clone from.
+   * @param target The path or table name to create the clone.
+   * @param replace Whether to replace the destination with the clone command.
+   *
+   * @since 3.3.0
+   */
+  def cloneAtTimestamp(timestamp: String, target: String, replace: Boolean): DeltaTable = {
+    cloneAtTimestamp(timestamp, target, replace, properties = Map.empty)
+  }
+
+  /**
+   * Clone a DeltaTable at a specific timestamp to a given destination to mirror the existing
+   * table's data and metadata.
+   * 
+   * Timestamp can be of the format yyyy-MM-dd or yyyy-MM-dd HH:mm:ss.
+   *
+   * @param timestamp The timestamp of this table to clone from.
+   * @param target The path or table name to create the clone.
+   * 
+   * An example would be
```

Comment:
The example should come before the `param`s.

Reply:
Copy pasta